src/Integration/ProjectPropertyManager.cs

```diff
@@ -0,0 +1,123 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ProjectPropertyManager.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using EnvDTE;
+using System;
+
+namespace SonarLint.VisualStudio.Integration
+{
+    [Export(typeof(IProjectPropertyManager))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal class ProjectPropertyManager : IProjectPropertyManager
+    {
+        public const string TestProperty = Constants.SonarQubeTestProjectBuildPropertyKey;
+        public const string ExcludeProperty = Constants.SonarQubeExcludeBuildPropertyKey;
+
+        private readonly IProjectSystemHelper projectSystem;
+
+        [ImportingConstructor]
+        public ProjectPropertyManager(IHost host)
+        {
+            if (host == null)
+            {
+                throw new ArgumentNullException(nameof(host));
+            }
+
+            this.projectSystem = host.GetService<IProjectSystemHelper>();
+            this.projectSystem.AssertLocalServiceIsNotNull();
+        }
+
+        internal /*for testing purposes*/ ProjectPropertyManager(IProjectSystemHelper projectSystem)
```

Comment:
Should not be needed since you can inject the project system helper you need, without adding test-only code.